src/components/erupted_item.rs

```diff
@@ -0,0 +1,49 @@
+use macroquad::math::{vec2, Vec2};
+use macroquad::prelude::{collections::storage, get_frame_time};
+
+use crate::Resources;
+
+use crate::components::PhysicsBody;
+
+//at the moment is not possible to test the code as ite requires EruptingVolcano to test
+//It also is dependency of ArmedGrenade so a rough estimation of the code to not implement the trait later
+//is here.
+pub trait EruptedItem {
```

Comment:
If the trait is not used anywhere - maybe exclude it from this PR? And add as a separate PR or withing the PR that is going to use it?

Reply:
Is "used" for the grenades, but the code can't be tested because it requires a functioning volcano. So for the separate PR, I think it should eliminated the trait from grenades.rs and put the version which has it implemented on the separate PR as you said.

Reply:
This PR is about Grenade item and EruptItems (especially untested) looks way out of scope of this PR as for

Reply:
done, moved away from the PR